designs/autoscaler.md

```diff
@@ -0,0 +1,526 @@
+
+EKS Anywhere builds on top of Cluster API (CAPI) and maintains several CAPI cloud providers for cluster provisioning and management.[Cluster Autoscaler on Cluster API](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/clusterapi) was chosen as the default EKS-A cluster autoscaling solution for its native CAPI support and easy integration, which avoids the need to drastically change the API definition and workflow. The following discussion will focus on Cluster Autoscaler as the autoscaling tool to integrate into EKS Anywhere.
+
+At launch, EKS Anywhere user can specify a fixed number of nodes for each worker node group in the cluster. Once a cluster is created, EKS-A cluster controller watches this configuration for changes, generates CAPI resources, and re-applies them in the cluster. Since EKS Anywhere does not support user interaction with CAPI directly, most manual CAPI resource changes made by the user will be overwritten by the EKS-A cluster controller. This behavior prevents users from enabling Cluster Autoscaler themselves, since the required autoscaling annotations manually applied to CAPI resources will be removed by EKS-A cluster controller.
+
+This document aims to solve this limitation and provides **autoscaling** **compatibility** in EKS Anywhere, so the user can either choose to install Cluster Autoscaler as an EKS-A curated package during or after cluster creation time, or install themselves from [upstream](https://github.com/kubernetes/autoscaler/releases) after a cluster is created.
+
+### Goals and Objectives
+
+As an EKS Anywhere user, I want to:
+
+* create clusters in a streamlined fashion with autoscaling capabilities
+* add autoscaling capabilities to existing EKS-A clusters
+* have clusters automatically scale worker nodes up and down within the set boundaries
+
+### Statement of Scope
+
+#### In Scope
+
+* Include Cluster Autoscaler as an [EKS-A curated package](https://anywhere.eks.amazonaws.com/docs/concepts/packages/).
+* Integrate Cluster Autoscaler to all CAPI providers since the feature is provider agnostic.
+* Integrate autoscaling configuration in cluster spec to make EKS-A cluster autoscaling-aware.
+* Generate an EKS-A opinionated Cluster Autoscaler curated package template with default configurations.
+* Install Cluster Autoscaler controller as a curated package during or after management/workload cluster creation.
+* Support upstream k8s Autoscaler installed/managed outside of EKS-A Curated Package.
+* Update the autoscaling configuration to set up the min/max number of nodes in each worker node group.
+
+#### Out of Scope
+
+* Compare cluster autoscaler products (e.g. [Karpenter](https://karpenter.sh/))
```

Comment:
Did we do some analysis on this before to determine to go with this native approach that we want to call out?

Reply:
internally, out of scope for this doc